File: aspnet-core/src/ERP.Healthcare.Application.Contracts/Doctors/CreateUpdateDoctorDto.cs
using System.ComponentModel.DataAnnotations;
using Volo.Abp.Application.Dtos;

namespace ERP.Healthcare.Doctors
{
    public class CreateUpdateDoctorDto : EntityDto<int>
    {
        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        [Required]
        [StringLength(100)]
        public string Description { get; set; }

        public GenderTypeEnum Gender { get; set; }

        public int AddressId { get; set; }

        public int DoctorSpecialtyId { get; set; }

        public int DoctorTitleId { get; set; }

        public CreateUpdateAddressDto Address { get; set; }
    }
}

File: angular/src/app/doctor/models/create-update-doctor-dto.ts

import { CreateUpdateAddressDto } from './create-update-address-dto';
import { GenderTypeEnum } from './gender-type-enum';

import { EntityDto } from '@abp/ng.core';

export class CreateUpdateDoctorDto extends EntityDto<number> {
  name: string;
  description: string;
  gender: any;
  addressId: number;
  doctorSpecialtyId: number;
  doctorTitleId: number;
  address: any;
  id: number;

  constructor(initialValues: Partial<CreateUpdateDoctorDto> = {}) {
    super(initialValues);
  }
}
